lib/generators/ruby_ui/component_generator.rb

```diff
@@ -90,20 +93,62 @@ def install_js_packages(js_packages)
       end
 
       def install_js_package(package)
-        if File.exist?(Rails.root.join("package-lock.json"))
-          run "npm install #{package}"
-        elsif File.exist?(Rails.root.join("yarn.lock"))
+        if using_importmap?
+          pin_with_importmap(package)
+        elsif using_yarn?
           run "yarn add #{package}"
+        elsif using_npm?
+          run "npm install #{package}"
         else
           say "Could not detect the package manager, you need to install '#{package}' manually", :red
         end
       end
 
+      def pin_with_importmap(package)
+        case package
+        when "motion"
+          pin_motion
+        when "tippy.js"
```

Comment:
These two packageds didn't play well with importmaps, so I'm pointing to remote cdn. My first take was to download the file and add to vendor/javascript folder and add `pin "motion"` to config/importmap.rb, but I'm not sure if that was the best option, because it could go wrong if user tried to update/redownload the packages. If think it's  safer to point to CDN and let the users download the file if they want.

Reply:
@stephannv , this is fine. Tippy.js will be replaced by floating-ui, and motion is used on the Accordion, but I don't think we need any dependency there. In other words, it's fine, and we should remove both dependencies soon.